### ISSUE TITLE
Add support for `JS` tracer to `muxTracer`

### ISSUE THREAD

**Issue by ArtificialPB** [paradigmxyz/evm-inspectors#58]
Follow-up of https://github.com/paradigmxyz/evm-inspectors/pull/57, currently `muxTracer` supports only built-in tracers. It should also support `JS` tracers.

In the alloy-rs tracing crate, the `MuxConfig` and `MuxFrame` types will need to have map key changed to `GethDebugTracerType`. Changes in this project should be pretty much the same, with addition of supporting the `JSInspector` via `DelegatingInspector`. Changes in `reth` are probably not needed.

The `JSInspector` is under `js-tracer` feature, so a feature gate for it in `MuxInspector` will need to be added as well.

**Comment by mattsse** [comment]
@ArtificialPB I'll need to bump evm-inspectors on reth shortly, so if you find the time I'd appreciate a pr that adds the missing `::Mux` handler for debug_trace* etc:

https://github.com/paradigmxyz/reth/blob/e21304947b6a46ba524d4b1e68351b05587d8892/crates/rpc/rpc/src/debug.rs#L279-L279

https://github.com/paradigmxyz/reth/blob/e21304947b6a46ba524d4b1e68351b05587d8892/crates/rpc/rpc/src/debug.rs#L512-L512

**Comment by ArtificialPB** [comment]
@mattsse yeah, just opened it about an hour ago: https://github.com/paradigmxyz/reth/pull/7053

**Comment by mattsse** [comment]
ah amazing!
ty

**Comment by kustrun** [comment]
@mattsse @ArtificialPB Opened 2 PRs to support `JS tracer` in the `muxTracer`

Alloy: https://github.com/alloy-rs/alloy/pull/2442
Revm-inspectors (dependent on Alloy changes): https://github.com/paradigmxyz/revm-inspectors/pull/291

What’s the best way to test these changes in `reth`, given that using local path overrides for dependencies results in `error[E0308]: mismatched types` during compilation?
